File: cordova/src_plugins/readbeyond-plugin-librarian/src/ios/Format.m
//
//            _           _            _ 
//           (_)         | |          | |
//  _ __ ___  _ _ __  ___| |_ _ __ ___| |
// | '_ ` _ \| | '_ \/ __| __| '__/ _ \ |
// | | | | | | | | | \__ \ |_| | |  __/ |
// |_| |_| |_|_|_| |_|___/\__|_|  \___|_|
//
// Author:      Alberto Pettarin (www.albertopettarin.it)
// Copyright:   Copyright 2013-2015, ReadBeyond Srl (www.readbeyond.it)
// License:     MIT
// Email:       minstrel@readbeyond.it
// Web:         http://www.readbeyond.it/minstrel/
// Status:      Production
//

#import "Format.h"

@implementation Format

- (id)init:(NSString *)name
{
    if (self == [super init]) {
        self.name     = name;
        self.version  = @"";
        self.metadata = [[NSMutableDictionary alloc] init];
    }
    return self;
}
- (id)init:(NSString *)name version:(NSString *)version
{
    if (self == [super init]) {
        self.name     = name;
        self.version  = @"";
        self.metadata = [[NSMutableDictionary alloc] init];
    }
    return self;
}

- (NSString *)getName
{
    return self.name;
}

- (void)addMetadatum:(NSString *)value forKey:(NSString *)name
{
    [self.metadata setValue:value forKey:name];
}

- (NSString *)getMetadatum:(NSString *)name
{
    return [self.metadata objectForKey:name];
}

- (BOOL)equals:(NSString *)otherName
{
    return [self.name isEqualToString:otherName];
}

- (BOOL)equals:(NSString *)otherName withVersion:(NSString *)otherVersion
{
    return [self.name isEqualToString:otherName] && [self.version isEqualToString:otherVersion];
}

- (id)toJSONObject
{
    NSMutableDictionary *obj = [[NSMutableDictionary alloc] init];
    [obj setValue:self.name     forKey:@"name"];
    [obj setValue:self.version  forKey:@"version"];
    [obj setValue:self.metadata forKey:@"metadata"];
    return obj;
}

@end

File: cordova/src_plugins/readbeyond-plugin-librarian/src/android/Format.java
//
//            _           _            _ 
//           (_)         | |          | |
//  _ __ ___  _ _ __  ___| |_ _ __ ___| |
// | '_ ` _ \| | '_ \/ __| __| '__/ _ \ |
// | | | | | | | | | \__ \ |_| | |  __/ |
// |_| |_| |_|_|_| |_|___/\__|_|  \___|_|
//
// Author:      Alberto Pettarin (www.albertopettarin.it)
// Copyright:   Copyright 2013-2015, ReadBeyond Srl (www.readbeyond.it)
// License:     MIT
// Email:       minstrel@readbeyond.it
// Web:         http://www.readbeyond.it/minstrel/
// Status:      Production
//

package it.readbeyond.minstrel.librarian;

import java.util.HashMap;
import java.util.Map.Entry;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class Format implements JSONPrintable {
   
    String name                      = "";                              // name
    String version                   = "";                              // version
    HashMap<String, String> metadata = new HashMap<String, String>();   // metadata

    public Format(String name, String version) {
        this.name    = name;
        this.version = version;
    }
    public Format(String name) {
        this(name, "");
    }

    public String getName() {
        return this.name;
    }
    public String getVersion() {
        return this.version;
    }

    public void setMetadata(HashMap<String, String> metadata) {
        this.metadata = metadata;
    }
    public HashMap<String, String> getMetadata() {
        return this.metadata;
    }
    public void addMetadatum(String name, String value) {
        this.metadata.put(name, value);
    }
    public String getMetadatum(String name) {
        return this.metadata.get(name);
    }

    public boolean equals(String otherName) {
        return this.name.equals(otherName);
    }
    public boolean equals(String otherName, String otherVersion) {
        return this.name.equals(otherName) && this.version.equals(otherVersion);
    }

    public JSONObject toJSONObject() {
        JSONObject obj =        new JSONObject();
        try {
            obj.put("name",     this.name);
            obj.put("version",  this.version);
            JSONObject meta =   new JSONObject();
            for (Entry<String, String> e : this.metadata.entrySet()) {
                meta.put(e.getKey(), e.getValue());
            }
            obj.put("metadata", meta);
        } catch (Exception e) {
            // nop
        }
        return obj;
    }
}
